src/pretix/eventyay_common/tasks.py

```diff
@@ -10,100 +10,153 @@
+)  # Retries up to 5 times with a 60-second delay
 def send_event_webhook(self, user_id, event, action):
     # Define the payload to send to the webhook
     user_model = get_user_model()
     user = user_model.objects.get(id=user_id)
     payload = {
-        'organiser_slug': event.get('organiser_slug'),
-        'name': event.get('name'),
-        'slug': event.get('slug'),
-        'date_from': event.get('date_from'),
-        'date_to': event.get('date_to'),
-        'timezone': event.get('timezone'),
-        'locale': event.get('locale'),
-        'locales': event.get('locales'),
-        'user_email': user.email,
-        'action': action
+        "organiser_slug": event.get("organiser_slug"),
+        "name": event.get("name"),
+        "slug": event.get("slug"),
+        "date_from": event.get("date_from"),
+        "date_to": event.get("date_to"),
+        "timezone": event.get("timezone"),
+        "locale": event.get("locale"),
+        "locales": event.get("locales"),
+        "user_email": user.email,
+        "action": action,
     }
     headers = get_header_token(user_id)
 
     try:
         # Send the POST request with the payload and the headers
-        response = requests.post(settings.TALK_HOSTNAME + '/webhook/event/',
-                                 json=payload,
-                                 headers=headers)
+        response = requests.post(
+            settings.TALK_HOSTNAME + "/webhook/event/", json=payload, headers=headers
+        )
         response.raise_for_status()  # Raise exception for bad status codes
     except requests.RequestException as e:
         # Log any errors that occur
-        logger.error('Error sending webhook to talk component: %s', e)
+        logger.error("Error sending webhook to talk component: %s", e)
         # Retry the task if an exception occurs (with exponential backoff by default)
         try:
             self.retry(exc=e)
         except self.MaxRetriesExceededError:
             logger.error("Max retries exceeded for sending organizer webhook.")
 
 
+@shared_task(
+    bind=True, max_retries=5, default_retry_delay=60
+)  # Retries up to 5 times with a 60-second delay
+def create_world(self, is_video_creation, data):
+    """
+    Create video system for the event
+    @self: task instance
+    @param is_video_creation: allow user to add video system
+    @param data: event's data
+    """
+    event_slug = data.get("id")
+    title = data.get("title")
+    event_timezone = data.get("timezone")
+    locale = data.get("locale")
+    token = data.get("token")
+    has_permission = data.get("has_permission")
+
+    payload = {
+        "id": event_slug,
+        "title": title,
+        "timezone": event_timezone,
+        "locale": locale,
+    }
+
+    headers = {"Authorization": "Bearer " + token}
+
+    # Check if user choose add video option and has permission to create video system ('can_create_events' permission)
+    if is_video_creation and has_permission:
```

Comment:
Please make `is_video_creation` and `has_permission` explicitly declared in 
`create_world()` signature.

```py
def create_world(self, is_video_creation, has_permission, data): ...
```

Doing so, if other people looking at its signature, they can know which parameters they need to pass when using this function.

Reply:
Please also resolve this.

Reply:
I added some comments for clarity in the last commit. If you need any further adjustments, please let me know. Thank you for reviewing! 
![Screenshot 2024-11-01 085212](https://github.com/user-attachments/assets/d6703ea3-72a1-41ef-8ddd-aa8173c5894d)